diff --git a/Cargo.toml b/Cargo.toml
index f29f281..05406b6 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -12,7 +12,7 @@ repository = "https://github.com/johanhelsing/bevy_pkv"
 [dependencies]
 thiserror = "1.0"
 serde = "1.0"
-bevy_ecs = { git = "https://github.com/bevyengine/bevy", optional = true } # we need for deriving Resource in PkvStore
+bevy_ecs = { git = "https://github.com/bevyengine/bevy" ,rev = "25f013ba1b809e9f962dafbefaeb860512f386e8",optional = true } # we need for deriving Resource in PkvStore
 
 [features]
 default = ["bevy", "sled"]
@@ -32,7 +32,7 @@ rmp-serde = "1.1"
 directories = "5.0"
 
 [dev-dependencies]
-bevy = { git = "https://github.com/bevyengine/bevy", default-features = false }
+bevy_ecs = { git = "https://github.com/bevyengine/bevy",rev = "25f013ba1b809e9f962dafbefaeb860512f386e8", default-features = false }
 strum_macros = "*"
 
 [build-dependencies]
@@ -60,4 +60,4 @@ required-features = ["bevy"]
 
 [[example]]
 name = "enumkeys"
-required-features = ["bevy"]
\ No newline at end of file
+required-features = ["bevy"]
diff --git a/src/lib.rs b/src/lib.rs
index 1a5705b..6f2d6ff 100644
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -124,7 +124,8 @@ impl PkvStore {
         self.inner.clear()
     }
 }
-
+#[cfg(bevy)]
+impl Resource for PkvStore{}
 struct PlatformDefault {
     qualifier: Option<String>,
     organization: String,
